File: jcu011/20220113/BOJ_20154.cpp
#include<bits/stdc++.h>
using namespace std;

int num[26]{};

int main() {
	ios_base::sync_with_stdio(0);
	cin.tie(0);
	num[0] = 3; num[1] = 2;
	num[2] = 1; num[3] = 2;
	num[4] = 3; num[5] = 3;
	num[6] = 3; num[7] = 3;
	num[8] = 1; num[9] = 1;
	num[10] = 3; num[11] = 1;
	num[12] = 3; num[13] = 3;
	num[14] = 1; num[15] = 2;
	num[16] = 2; num[17] = 2;
	num[18] = 1; num[19] = 2;
	num[20] = 1; num[21] = 1;
	num[22] = 2; num[23] = 2;
	num[24] = 2; num[25] = 1;

	string str;
	cin>>str;

	int sum=0;
	for(int i=0; i<str.size(); i++){
		sum += num[str[i]-'A'];
	}

	if((sum%10)%2)cout<<"I'm a winner!";
	else cout<<"You're the winner?";

}

File: choiyoungkwon12/20220107/BOJ_20154.java
package BOJ;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Deque;
import java.util.LinkedList;

/**
 * https://www.acmicpc.net/problem/20154
 */

public class BOJ_20154 {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String str = br.readLine();

        int[] num = new int[]{3, 2, 1, 2, 3, 3, 3, 3, 1, 1, 3, 1, 3, 3, 1, 2, 2, 2, 1, 2, 1, 1, 2, 2, 2, 1};

        Deque<Integer> queue = new LinkedList<>();

        for (int i = 0; i < str.length(); i++) {
            int number = num[str.charAt(i) - 'A'];
            queue.add(number);
        }

        while (queue.size() >= 2) {
            // 큐에 있는거 2개 빼서 더한 후 다시 큐에 넣기 반복
            Integer pop = queue.pop();
            Integer pop1 = queue.pop();
            int sum = pop + pop1;
            if (sum >= 10) {
                sum %= 10;
            }
            queue.push(sum);
        }

        Integer result = queue.pop();

        if (result % 2 == 0){
            System.out.println("You're the winner?");
        }else{
            System.out.println("I'm a winner!");
        }
    }
}
